### PR TITLE
OCPBUGS-2666: Add RBAC check on Create a Project link in all-namespaces pages

### PR DESCRIPTION
**Pull request by vikram-raj**
Fixes: https://issues.redhat.com/browse/OCPBUGS-2666

Descriptions: Add RBAC check on Create a Project link in all-namespaces pages in dev console

### CODE REVIEW

File: frontend/packages/dev-console/src/components/projects/CreateProjectListPage.tsx [hunk 1]
```diff
@@ -11,6 +13,18 @@ export interface CreateProjectListPageProps extends ProjectListPageProps {
...
+  return (
+    <Trans t={t} ns="devconsole">
+      or{' '}
+      <Button isInline variant="link" onClick={openProjectModal}>
+        create a Project
+      </Button>
+    </Trans>
+  );
+};
+
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/projects/CreateProjectListPage.tsx:27]: Below in `CreateProjectListPage` the `openProjectModal` callback is created and passed to the `CreateProjectListPage` `children`...

Just an idea, don't know if this works fine. You might be also can check the flag here and pass the button as optional the child? Wdyt?

```tsx
const canCreateProject = useFlag(FLAGS.CAN_CREATE_PROJECT);
const orProjectButton = canCreateProject && (
    <Trans t={t} ns="devconsole">
      {' or '}
      <Button isInline variant="link" onClick={openProjectModal}>
        create a Project
      </Button>
    </Trans>
)
```

and then:

```tsx
  return (
    <ProjectListPage {...props} title={title}>
      {children({ orProjectButton })}
    </ProjectListPage>
  );
```

Just an idea. If you prefer the current impl. that's fine for me. But we need to address the translation issues. (Unnecessary space and the full stop should be part of the translations.)

File: frontend/packages/dev-console/src/components/builds/BuildsTabListPage.tsx [hunk 2]
```diff
@@ -96,13 +96,18 @@ const BuildsTabListPage: React.FC<BuildsTabListPageProps> = ({ match }) => {
...
+          <>
+            {t('devconsole~Select a Project to view the list of builds')}{' '}
+            {canCreateProject && (
+              <Trans t={t} ns="devconsole">
+                or{' '}
+                <Button isInline variant="link" onClick={openProjectModal}>
+                  create a Project
+                </Button>
+              </Trans>
+            )}
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/builds/BuildsTabListPage.tsx:108]: I believe you missed using your new component here, or is there a difference that I didn't see?

File: frontend/packages/dev-console/src/components/catalog/CatalogPage.tsx [hunk 3]
```diff
@@ -50,13 +53,18 @@ const CatalogPage: React.FC<CatalogPageProps> = ({ match }) => {
...
-              </Button>
+            <>
+              {t('devconsole~Select a Project to view the developer catalog')}{' '}
+              {canCreateProject && (
+                <Trans t={t} ns="devconsole">
+                  or{' '}
+                  <Button isInline variant="link" onClick={openProjectModal}>
+                    create a Project
+                  </Button>
+                </Trans>
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/catalog/CatalogPage.tsx:64]: I believe you missed using your new component here, or is there a difference that I didn't see?

File: frontend/packages/dev-console/src/components/import/SamplesCatalog.tsx [hunk 4]
```diff
@@ -34,13 +35,18 @@ const SampleCatalog: React.FC<SampleCatalogProps> = ({ match }) => {
...
-                </Button>
+              <>
+                {t('devconsole~Select a Project to view the list of samples')}{' '}
+                {canCreateProject && (
+                  <Trans t={t} ns="devconsole">
+                    or{' '}
+                    <Button isInline variant="link" onClick={openProjectModal}>
+                      create a Project
+                    </Button>
+                  </Trans>
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/import/SamplesCatalog.tsx:46]: I believe you missed using your new component here, or is there a difference that I didn't see?

File: frontend/packages/dev-console/src/components/ProjectSelectPage.tsx [hunk 5]
```diff
@@ -56,11 +55,8 @@ const ProjectSelectPage: React.FC<ProjectSelectPageProps> = (props) => {
       <CreateProjectListPage title={kindObj.labelPlural} badge={getBadgeFromType(kindObj.badge)}>
         {(openProjectModal) => (
           <Trans t={t} ns="devconsole" values={{ projectLabelPlural }}>
-            Select a Project to view the list of {{ projectLabelPlural }} or{' '}
-            <Button isInline variant="link" onClick={openProjectModal}>
-              create a Project
-            </Button>
-            .
+            elect a Project to view the list of {{ projectLabelPlural }}
```
Comment (jerolimov) [inline frontend/packages/dev-console/src/components/ProjectSelectPage.tsx:58]: Nit :smirk: 

```suggestion
            Select a Project to view the list of {{ projectLabelPlural }}
```

**Review by jerolimov — CHANGES_REQUESTED** [review]
Tested it with a cluster bot and removed the create project permission from `ClusterRole` `self-provisioner`, works fine.

But sorry, there is one S missing. :see_no_evil:

File: frontend/packages/dev-console/src/components/ProjectSelectPage.tsx (hunk 5 above)
Comment (vikram-raj) [reply frontend/packages/dev-console/src/components/ProjectSelectPage.tsx]: My bad. fixed it

**Review by jerolimov — APPROVED** [review]
/lgtm
/cherry-pick release-4.12

**Comment by vikram-raj** [comment]
Manual cherry-pick https://github.com/openshift/console/pull/12489

**Pull request merged**